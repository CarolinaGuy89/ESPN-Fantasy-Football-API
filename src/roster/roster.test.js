import _ from 'lodash';

import BaseAPIObject from '../base-api-object/base-api-object.js';
import SlottedPlayer from '../slotted-player/slotted-player.js';
import Team from '../team/team.js';

import Roster from './roster.js';

import { localObject, serverResponse } from './roster.stubs.js';

describe('Roster', () => {
  let roster;

  beforeEach(() => {
    roster = new Roster();
  });

  afterEach(() => {
    roster = null;
  });

  test('extends BaseAPIObject', () => {
    expect(roster).toBeInstanceOf(BaseAPIObject);
  });

  describe('attribute population from server response', () => {
    beforeEach(() => {
      roster = Roster.buildFromServer(serverResponse, { teamId: 9 });
    });

    test('parses data correctly', () => {
      expect(roster).toMatchSnapshot();
    });
  });

  describe('attribute population from local object', () => {
    beforeEach(() => {
      roster = new Roster(localObject);
    });

    test('parses data correctly', () => {
      expect(roster).toMatchSnapshot();
    });
  });

  describe('constructor', () => {
    describe('when options are not passed', () => {
      const testPropIsUndefined = (prop) => {
        test(`${prop} is undefined`, () => {
          const newRoster = new Roster();
          expect(_.get(newRoster, prop)).toBeUndefined();
        });
      };

      testPropIsUndefined('leagueId');
      testPropIsUndefined('seasonId');
      testPropIsUndefined('teamId');
      testPropIsUndefined('scoringPeriodId');
    });

    describe('when options are passed', () => {
      const testPropIsSetFromOptions = (prop) => {
        test(`${prop} is set from options`, () => {
          const value = 25;
          const newRoster = new Roster({ [prop]: value });
          expect(_.get(newRoster, prop)).toBe(value);
        });
      };

      testPropIsSetFromOptions('leagueId');
      testPropIsSetFromOptions('seasonId');
      testPropIsSetFromOptions('teamId');
      testPropIsSetFromOptions('scoringPeriodId');
    });
  });

  describe('responseMap', () => {
    describe('team', () => {
      describe('manualParse', () => {
        let instance;

        beforeEach(() => {
          instance = new Roster({
            leagueId: 123,
            seasonId: 2018
          });
        });

        afterEach(() => {
          instance = null;
        });

        const testReturnsEmptyInstance = ({ value, valueString }) => {
          describe(`when the passed data is ${valueString}`, () => {
            test('returns an empty Team', () => {
              const expectedTeam = new Team({
                leagueId: instance.leagueId, seasonId: instance.seasonId
              });

              const returnedTeam = Roster.responseMap.team.manualParse(value, undefined, instance);
              expect(returnedTeam).toEqual(expectedTeam);
            });
          });
        };

        describe('when the populating instance does not have a teamId', () => {
          beforeEach(() => {
            instance.teamId = undefined;
          });

          testReturnsEmptyInstance({ value: undefined, valueString: 'undefined' });
          testReturnsEmptyInstance({ value: null, valueString: 'null' });
          testReturnsEmptyInstance({ value: [], valueString: 'empty array' });
          testReturnsEmptyInstance({
            value: [{ teamId: 4 }, { teamId: 5 }, { teamId: 6 }],
            valueString: 'populated array'
          });
        });

        describe('when the populating instance has a teamId', () => {
          beforeEach(() => {
            instance.teamId = 10;
          });

          testReturnsEmptyInstance({ value: undefined, valueString: 'undefined' });
          testReturnsEmptyInstance({ value: null, valueString: 'null' });
          testReturnsEmptyInstance({ value: [], valueString: 'empty array' });

          describe('when the passed data is a populated array', () => {
            let responseData;

            beforeEach(() => {
              responseData = [{
                teamId: instance.teamId,
                team: {
                  teamId: instance.teamId
                }
              }];
            });

            afterEach(() => {
              responseData = null;
            });

            describe('when there is a cached team', () => {
              test('returns the cached team', () => {
                const cachedTeam = Team.buildFromServer(
                  { teamId: instance.teamId },
                  { leagueId: instance.leagueId, seasonId: instance.seasonId }
                );

                const returnedTeam = Roster.responseMap.team.manualParse(
                  responseData, undefined, instance
                );
                expect(returnedTeam).toBe(cachedTeam);

                Team.clearCache();
              });
            });

            describe('when there is not a cached team', () => {
              test('creates a new team', () => {
                Team.clearCache();

                const cachingId = Team.getCacheId({
                  leagueId: instance.leagueId,
                  seasonId: instance.seasonId,
                  teamId: instance.teamId
                });

                const returnedTeam = Roster.responseMap.team.manualParse(
                  responseData, undefined, instance
                );
                expect(returnedTeam).toBe(Team.get(cachingId));

                Team.clearCache();
              });
            });
          });
        });
      });
    });

    describe('player', () => {
      describe('manualParse', () => {
        let instance;

        beforeEach(() => {
          instance = new Roster({
            leagueId: 123,
            seasonId: 2018
          });
        });

        afterEach(() => {
          instance = null;
        });

        const testReturnsEmptyArray = ({ value, valueString }) => {
          describe(`when the passed data is ${valueString}`, () => {
            test('returns an empty array', () => {
              const returnedPlayers = Roster.responseMap.players.manualParse(
                value, undefined, instance
              );
              expect(returnedPlayers).toEqual([]);
            });
          });
        };

        describe('when the populating instance does not have a teamId', () => {
          beforeEach(() => {
            instance.teamId = undefined;
          });

          testReturnsEmptyArray({ value: undefined, valueString: 'undefined' });
          testReturnsEmptyArray({ value: null, valueString: 'null' });
          testReturnsEmptyArray({ value: [], valueString: 'empty array' });
          testReturnsEmptyArray({
            value: [{ teamId: 4 }, { teamId: 5 }, { teamId: 6 }],
            valueString: 'populated array'
          });
        });

        describe('when the populating instance has a teamId', () => {
          beforeEach(() => {
            instance.teamId = 10;
          });

          testReturnsEmptyArray({ value: undefined, valueString: 'undefined' });
          testReturnsEmptyArray({ value: null, valueString: 'null' });
          testReturnsEmptyArray({ value: [], valueString: 'empty array' });

          describe('when the passed data is a populated array', () => {
            test('returns an array of SlottedPlayers', () => {
              const responseData = [{
                teamId: instance.teamId,
                slots: [{
                  player: {},
                  isKeeper: true
                }, {
                  player: {},
                  isKeeper: false
                }]
              }];

              const returnedPlayers = Roster.responseMap.players.manualParse(
                responseData, undefined, instance
              );

              expect.hasAssertions();
              _.forEach(returnedPlayers, (slottedPlayer) => {
                expect(slottedPlayer).toBeInstanceOf(SlottedPlayer);
              });
            });
          });
        });
      });
    });
  });

  describe('class methods', () => {
    describe('getCacheId', () => {
      let leagueId, seasonId, teamId, scoringPeriodId;

      afterEach(() => {
        leagueId = null;
        seasonId = null;
        teamId = null;
        scoringPeriodId = null;
      });

      const testReturnsUndefined = () => {
        test('returns undefined', () => {
          const params = { leagueId, seasonId, teamId, scoringPeriodId };
          expect(Roster.getCacheId(params)).toBeUndefined();
        });
      };

      describe('when called with no params', () => {
        test('returns undefined', () => {
          expect(Roster.getCacheId()).toBeUndefined();
        });
      });

      describe('when leagueId is defined', () => {
        beforeEach(() => {
          leagueId = 132123;
        });

        describe('when seasonId is defined', () => {
          beforeEach(() => {
            seasonId = 2017;
          });

          describe('when teamId is defined', () => {
            beforeEach(() => {
              teamId = 4;
            });

            describe('when scoringPeriodId is defined', () => {
              beforeEach(() => {
                scoringPeriodId = 11;
              });

              test('returns a valid caching id', () => {
                const params = { leagueId, seasonId, teamId, scoringPeriodId };

                const returnedCachingId = Roster.getCacheId(params);
                expect(returnedCachingId).toBe(
                  `${teamId}-${leagueId}-${seasonId}-${scoringPeriodId}`
                );
              });
            });

            describe('when scoringPeriodId is undefined', () => {
              beforeEach(() => {
                scoringPeriodId = undefined;
              });

              testReturnsUndefined();
            });
          });

          describe('when teamId is undefined', () => {
            beforeEach(() => {
              teamId = undefined;
            });

            describe('when scoringPeriodId is defined', () => {
              beforeEach(() => {
                scoringPeriodId = 11;
              });

              testReturnsUndefined();
            });

            describe('when scoringPeriodId is undefined', () => {
              beforeEach(() => {
                scoringPeriodId = undefined;
              });

              testReturnsUndefined();
            });
          });
        });

        describe('when seasonId is undefined', () => {
          beforeEach(() => {
            seasonId = undefined;
          });

          describe('when teamId is defined', () => {
            beforeEach(() => {
              teamId = 4;
            });

            describe('when scoringPeriodId is defined', () => {
              beforeEach(() => {
                scoringPeriodId = 11;
              });

              testReturnsUndefined();
            });

            describe('when scoringPeriodId is undefined', () => {
              beforeEach(() => {
                scoringPeriodId = undefined;
              });

              testReturnsUndefined();
            });
          });

          describe('when teamId is undefined', () => {
            beforeEach(() => {
              teamId = undefined;
            });

            describe('when scoringPeriodId is defined', () => {
              beforeEach(() => {
                scoringPeriodId = 11;
              });

              testReturnsUndefined();
            });

            describe('when scoringPeriodId is undefined', () => {
              beforeEach(() => {
                scoringPeriodId = undefined;
              });

              testReturnsUndefined();
            });
          });
        });
      });

      describe('when leagueId is undefined', () => {
        beforeEach(() => {
          leagueId = undefined;
        });

        describe('when seasonId is defined', () => {
          beforeEach(() => {
            seasonId = 2017;
          });

          describe('when teamId is defined', () => {
            beforeEach(() => {
              teamId = 4;
            });

            describe('when scoringPeriodId is defined', () => {
              beforeEach(() => {
                scoringPeriodId = 11;
              });

              testReturnsUndefined();
            });

            describe('when scoringPeriodId is undefined', () => {
              beforeEach(() => {
                scoringPeriodId = undefined;
              });

              testReturnsUndefined();
            });
          });

          describe('when teamId is undefined', () => {
            beforeEach(() => {
              teamId = undefined;
            });

            describe('when scoringPeriodId is defined', () => {
              beforeEach(() => {
                scoringPeriodId = 11;
              });

              testReturnsUndefined();
            });

            describe('when scoringPeriodId is undefined', () => {
              beforeEach(() => {
                scoringPeriodId = undefined;
              });

              testReturnsUndefined();
            });
          });
        });

        describe('when seasonId is undefined', () => {
          beforeEach(() => {
            seasonId = undefined;
          });

          describe('when teamId is defined', () => {
            beforeEach(() => {
              teamId = 4;
            });

            describe('when scoringPeriodId is defined', () => {
              beforeEach(() => {
                scoringPeriodId = 11;
              });

              testReturnsUndefined();
            });

            describe('when scoringPeriodId is undefined', () => {
              beforeEach(() => {
                scoringPeriodId = undefined;
              });

              testReturnsUndefined();
            });
          });

          describe('when teamId is undefined', () => {
            beforeEach(() => {
              teamId = undefined;
            });

            describe('when scoringPeriodId is defined', () => {
              beforeEach(() => {
                scoringPeriodId = 11;
              });

              testReturnsUndefined();
            });

            describe('when scoringPeriodId is undefined', () => {
              beforeEach(() => {
                scoringPeriodId = undefined;
              });

              testReturnsUndefined();
            });
          });
        });
      });
    });

    describe('read', () => {
      describe('when nothing is passed to read', () => {
        test('throws error', () => {
          expect(() => Roster.read()).toThrowError(
            `${Roster.displayName}: static read: cannot read without leagueId`
          );
        });
      });

      describe('when params are passed to read', () => {
        const testDefersRead = ({ params, expectedParams }) => {
          test('defers to super.read', () => {
            jest.spyOn(BaseAPIObject, 'read').mockImplementation();

            const instance = new Roster();
            const route = 'some route';
            const reload = false;

            Roster.read({ instance, route, params, reload });
            expect(BaseAPIObject.read).toBeCalledWith({
              instance, route, params: expectedParams, reload
            });

            BaseAPIObject.read.mockRestore();
          });
        };

        const testThrowsError = ({ params, errorMessage }) => {
          test('throws error', () => {
            expect(() => Roster.read({ params })).toThrowError(errorMessage);
          });
        };

        describe('when leagueId is passed on params', () => {
          describe('when seasonId is passed on params', () => {
            describe('when teamIds is passed on params', () => {
              describe('when teamId is passed on params', () => {
                const params = { leagueId: 3213, seasonId: 2017, teamId: 9, teamIds: 9 };
                testDefersRead({
                  params,
                  expectedParams: params
                });
              });

              describe('when teamId is not passed on params', () => {
                const params = { leagueId: 3213, seasonId: 2017, teamIds: 9 };
                testDefersRead({
                  params,
                  expectedParams: params
                });
              });
            });

            describe('when teamIds is not passed on params', () => {
              describe('when teamId is passed on params', () => {
                testDefersRead({
                  params: { leagueId: 3213, seasonId: 2017, teamId: 9 },
                  expectedParams: { leagueId: 3213, seasonId: 2017, teamId: 9, teamIds: 9 }
                });
              });

              describe('when teamId is not passed on params', () => {
                testThrowsError({
                  params: { leagueId: 1231232, seasonId: 2017 },
                  errorMessage: 'Roster: static read: cannot read without teamId'
                });
              });
            });
          });

          describe('when seasonId is not passed on params', () => {
            describe('when teamIds is passed on params', () => {
              describe('when teamId is passed on params', () => {
                testThrowsError({
                  params: { leagueId: 1231232, teamId: 9, teamIds: 9 },
                  errorMessage: 'Roster: static read: cannot read without seasonId'
                });
              });

              describe('when teamId is not passed on params', () => {
                testThrowsError({
                  params: { leagueId: 1231232, teamIds: 9 },
                  errorMessage: 'Roster: static read: cannot read without seasonId'
                });
              });
            });

            describe('when teamIds is not passed on params', () => {
              describe('when teamId is passed on params', () => {
                testThrowsError({
                  params: { leagueId: 1231232, teamId: 9 },
                  errorMessage: 'Roster: static read: cannot read without seasonId'
                });
              });

              describe('when teamId is not passed on params', () => {
                testThrowsError({
                  params: { leagueId: 1231232 },
                  errorMessage: 'Roster: static read: cannot read without seasonId'
                });
              });
            });
          });
        });

        describe('when leagueId is not passed on params', () => {
          describe('when seasonId is passed on params', () => {
            describe('when teamIds is passed on params', () => {
              describe('when teamId is passed on params', () => {
                testThrowsError({
                  params: { seasonId: 2017, teamId: 9, teamIds: 9 },
                  errorMessage: 'Roster: static read: cannot read without leagueId'
                });
              });

              describe('when teamId is not passed on params', () => {
                testThrowsError({
                  params: { seasonId: 2017, teamIds: 9 },
                  errorMessage: 'Roster: static read: cannot read without leagueId'
                });
              });
            });

            describe('when teamIds is not passed on params', () => {
              describe('when teamId is passed on params', () => {
                testThrowsError({
                  params: { seasonId: 2017, teamId: 9 },
                  errorMessage: 'Roster: static read: cannot read without leagueId'
                });
              });

              describe('when teamId is not passed on params', () => {
                testThrowsError({
                  params: { seasonId: 2017 },
                  errorMessage: 'Roster: static read: cannot read without leagueId'
                });
              });
            });
          });

          describe('when seasonId is not passed on params', () => {
            describe('when teamIds is passed on params', () => {
              describe('when teamId is passed on params', () => {
                testThrowsError({
                  params: { teamId: 9, teamIds: 9 },
                  errorMessage: 'Roster: static read: cannot read without leagueId'
                });
              });

              describe('when teamId is not passed on params', () => {
                testThrowsError({
                  params: { teamIds: 9 },
                  errorMessage: 'Roster: static read: cannot read without leagueId'
                });
              });
            });

            describe('when teamIds is not passed on params', () => {
              describe('when teamId is passed on params', () => {
                testThrowsError({
                  params: { teamId: 9 },
                  errorMessage: 'Roster: static read: cannot read without leagueId'
                });
              });

              describe('when teamId is not passed on params', () => {
                testThrowsError({
                  params: {},
                  errorMessage: 'Roster: static read: cannot read without leagueId'
                });
              });
            });
          });
        });
      });
    });
  });

  describe('instance methods', () => {
    describe('read', () => {
      beforeEach(() => {
        jest.spyOn(BaseAPIObject.prototype, 'read').mockImplementation();
      });

      afterEach(() => {
        BaseAPIObject.prototype.read.mockRestore();
      });

      describe('when params are passed to the method', () => {
        describe('when id params are defined on the instance', () => {
          test('calls super.read with only defined id params', () => {
            const instance = new Roster({
              leagueId: 4213,
              seasonId: 2018,
              teamId: 4,
              scoringPeriodId: 12
            });
            const params = { some: 'params' };
            const reload = false;

            instance.read({ params, reload });
            expect(BaseAPIObject.prototype.read).toBeCalledWith({
              params: _.assign({}, params, {
                leagueId: instance.leagueId,
                seasonId: instance.seasonId,
                teamIds: instance.teamId,
                scoringPeriodId: instance.scoringPeriodId
              }),
              route: Roster.route,
              reload
            });
          });
        });

        describe('when id params are undefined on the instance', () => {
          test('calls super.read with only defined id params', () => {
            const instance = new Roster();
            const params = { some: 'params' };
            const route = 'some route';

            instance.read({ params, route });
            expect(BaseAPIObject.prototype.read).toBeCalledWith({
              params,
              route,
              reload: true
            });
          });
        });
      });

      describe('when no params are passed to the method', () => {
        describe('when id params are defined on the instance', () => {
          test('calls super.read with only defined id params', () => {
            const instance = new Roster({
              leagueId: 4213,
              seasonId: 2018,
              teamId: 4,
              scoringPeriodId: 12
            });

            instance.read();
            expect(BaseAPIObject.prototype.read).toBeCalledWith({
              params: {
                leagueId: instance.leagueId,
                seasonId: instance.seasonId,
                teamIds: instance.teamId,
                scoringPeriodId: instance.scoringPeriodId
              },
              route: Roster.route,
              reload: true
            });
          });
        });

        describe('when id params are undefined on the instance', () => {
          test('calls super.read with only defined id params', () => {
            const instance = new Roster();

            instance.read();
            expect(BaseAPIObject.prototype.read).toBeCalledWith({
              params: {},
              route: Roster.route,
              reload: true
            });
          });
        });
      });
    });
  });
});
