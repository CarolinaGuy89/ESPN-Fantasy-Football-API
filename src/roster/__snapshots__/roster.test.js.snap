// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Roster when creating a team from a server response parses and assigns data correctly 1`] = `
Roster {
  "leagueId": undefined,
  "players": Array [
    SlottedPlayer {
      "isKeeper": false,
      "isLocked": true,
      "player": Player {
        "eligiblePositions": Array [
          "QB",
          "Bench",
        ],
        "firstName": "Tom",
        "isActive": true,
        "isIREligible": false,
        "jerseyNumber": "12",
        "lastName": "Brady",
        "percentOwned": 99.80315,
        "percentOwnedChange": 0.20315,
        "percentStarted": 81.69291,
        "playerId": 2330,
      },
      "position": "QB",
    },
    SlottedPlayer {
      "isKeeper": false,
      "isLocked": true,
      "player": Player {
        "eligiblePositions": Array [
          "RB",
          "RB/WR/TE",
          "Bench",
        ],
        "firstName": "Jerick",
        "isActive": true,
        "isIREligible": false,
        "jerseyNumber": "21",
        "lastName": "McKinnon",
        "percentOwned": 87.4613,
        "percentOwnedChange": -1.08439,
        "percentStarted": 50.77399,
        "playerId": 16782,
      },
      "position": "RB",
    },
    SlottedPlayer {
      "isKeeper": false,
      "isLocked": true,
      "player": Player {
        "eligiblePositions": Array [
          "RB",
          "RB/WR/TE",
          "Bench",
        ],
        "firstName": "Malcolm",
        "isActive": true,
        "isIREligible": false,
        "jerseyNumber": "34",
        "lastName": "Brown",
        "percentOwned": 15.63574,
        "percentOwnedChange": -0.20277,
        "percentStarted": 2.06186,
        "playerId": 17742,
      },
      "position": "RB",
    },
    SlottedPlayer {
      "isKeeper": false,
      "isLocked": true,
      "player": Player {
        "eligiblePositions": Array [
          "WR",
          "RB/WR/TE",
          "Bench",
        ],
        "firstName": "Golden",
        "isActive": true,
        "isIREligible": false,
        "jerseyNumber": "15",
        "lastName": "Tate",
        "percentOwned": 95.61404,
        "percentOwnedChange": 0.53574,
        "percentStarted": 60.74561,
        "playerId": 13217,
      },
      "position": "WR",
    },
    SlottedPlayer {
      "isKeeper": false,
      "isLocked": true,
      "player": Player {
        "eligiblePositions": Array [
          "WR",
          "RB/WR/TE",
          "Bench",
        ],
        "firstName": "Adam",
        "isActive": true,
        "isIREligible": false,
        "jerseyNumber": "19",
        "lastName": "Thielen",
        "percentOwned": 96.88073,
        "percentOwnedChange": -0.06842,
        "percentStarted": 82.20183,
        "playerId": 17178,
      },
      "position": "WR",
    },
    SlottedPlayer {
      "isKeeper": false,
      "isLocked": true,
      "player": Player {
        "eligiblePositions": Array [
          "TE",
          "RB/WR/TE",
          "Bench",
        ],
        "firstName": "Kyle",
        "isActive": true,
        "isIREligible": false,
        "jerseyNumber": "82",
        "lastName": "Rudolph",
        "percentOwned": 90.86957,
        "percentOwnedChange": 0.18131,
        "percentStarted": 59.56522,
        "playerId": 14054,
      },
      "position": "TE",
    },
    SlottedPlayer {
      "isKeeper": false,
      "isLocked": true,
      "player": Player {
        "eligiblePositions": Array [
          "WR",
          "RB/WR/TE",
          "Bench",
        ],
        "firstName": "Ted",
        "isActive": true,
        "isIREligible": false,
        "jerseyNumber": "19",
        "lastName": "Ginn Jr.",
        "percentOwned": 47.53363,
        "percentOwnedChange": -1.37726,
        "percentStarted": 8.07175,
        "playerId": 10453,
      },
      "position": "RB/WR/TE",
    },
    SlottedPlayer {
      "isKeeper": false,
      "isLocked": true,
      "player": Player {
        "eligiblePositions": Array [
          "D/ST",
          "Bench",
        ],
        "firstName": "Patriots",
        "isActive": true,
        "isIREligible": false,
        "lastName": "D/ST",
        "percentOwned": 56.66667,
        "percentOwnedChange": 0.65644,
        "percentStarted": 33.88889,
        "playerId": 60017,
      },
      "position": "D/ST",
    },
    SlottedPlayer {
      "isKeeper": false,
      "isLocked": true,
      "player": Player {
        "eligiblePositions": Array [
          "K",
          "Bench",
        ],
        "firstName": "Stephen",
        "isActive": true,
        "isIREligible": false,
        "jerseyNumber": "3",
        "lastName": "Gostkowski",
        "percentOwned": 98.5258,
        "percentOwnedChange": -0.26452,
        "percentStarted": 87.22359,
        "playerId": 9704,
      },
      "position": "K",
    },
    SlottedPlayer {
      "isKeeper": false,
      "isLocked": true,
      "player": Player {
        "eligiblePositions": Array [
          "RB",
          "RB/WR/TE",
          "Bench",
        ],
        "firstName": "Todd",
        "isActive": true,
        "isIREligible": false,
        "jerseyNumber": "30",
        "lastName": "Gurley II",
        "percentOwned": 98.57143,
        "percentOwnedChange": -0.06803,
        "percentStarted": 90.15873,
        "playerId": 17683,
      },
      "position": "Bench",
    },
    SlottedPlayer {
      "isKeeper": false,
      "isLocked": true,
      "player": Player {
        "eligiblePositions": Array [
          "WR",
          "RB/WR/TE",
          "Bench",
        ],
        "firstName": "Tyreek",
        "isActive": true,
        "isIREligible": false,
        "jerseyNumber": "10",
        "lastName": "Hill",
        "percentOwned": 95.94156,
        "percentOwnedChange": -0.11107,
        "percentStarted": 72.56494,
        "playerId": 17929,
      },
      "position": "Bench",
    },
    SlottedPlayer {
      "isKeeper": false,
      "isLocked": true,
      "player": Player {
        "eligiblePositions": Array [
          "TE",
          "RB/WR/TE",
          "Bench",
        ],
        "firstName": "Zach",
        "isActive": true,
        "isIREligible": false,
        "jerseyNumber": "86",
        "lastName": "Ertz",
        "percentOwned": 97.42647,
        "percentOwnedChange": -0.20386,
        "percentStarted": 83.08824,
        "playerId": 15835,
      },
      "position": "Bench",
    },
    SlottedPlayer {
      "isKeeper": false,
      "isLocked": true,
      "player": Player {
        "eligiblePositions": Array [
          "K",
          "Bench",
        ],
        "firstName": "Harrison",
        "isActive": true,
        "isIREligible": false,
        "jerseyNumber": "7",
        "lastName": "Butker",
        "percentOwned": 59.9297,
        "percentOwnedChange": 1.14481,
        "percentStarted": 49.56063,
        "playerId": 18409,
      },
      "position": "Bench",
    },
    SlottedPlayer {
      "isKeeper": false,
      "isLocked": true,
      "player": Player {
        "eligiblePositions": Array [
          "QB",
          "Bench",
        ],
        "firstName": "Matthew",
        "isActive": true,
        "isIREligible": false,
        "jerseyNumber": "9",
        "lastName": "Stafford",
        "percentOwned": 90.16787,
        "percentOwnedChange": -0.79599,
        "percentStarted": 51.55875,
        "playerId": 12483,
      },
      "position": "Bench",
    },
    SlottedPlayer {
      "isKeeper": false,
      "isLocked": true,
      "player": Player {
        "eligiblePositions": Array [
          "RB",
          "RB/WR/TE",
          "Bench",
        ],
        "firstName": "Jonathan",
        "isActive": true,
        "isIREligible": false,
        "jerseyNumber": "28",
        "lastName": "Stewart",
        "percentOwned": 58.45411,
        "percentOwnedChange": -0.24155,
        "percentStarted": 12.31884,
        "playerId": 11247,
      },
      "position": "Bench",
    },
    SlottedPlayer {
      "position": "Bench",
    },
  ],
  "scoringPeriodId": undefined,
  "seasonId": undefined,
  "team": Team {
    "abbreviation": "��",
    "division": Object {
      "divisionId": 0,
      "divisionName": "Division 1",
      "size": 4,
    },
    "divisionStanding": 2,
    "firstName": "2 Gurleys",
    "lastName": "One Cup",
    "leagueId": undefined,
    "leagueStanding": 3,
    "logoURL": "http://i.imgur.com/C17g1Af.gif",
    "seasonId": undefined,
    "streakType": "ERROR: streakType not recognized",
    "teamId": 9,
    "waiverRank": 2,
  },
  "teamId": 9,
}
`;

exports[`Roster when creating a team locally parses and assigns data correctly 1`] = `
Roster {
  "leagueId": undefined,
  "players": Array [],
  "scoringPeriodId": undefined,
  "seasonId": undefined,
  "team": Object {},
  "teamId": undefined,
}
`;
